tests/unit/plugins/inventory/test_kubevirt_composable_vars.py

```diff
@@ -0,0 +1,153 @@
+# Copyright: (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import pytest
+
+from ansible_collections.kubevirt.core.plugins.inventory.kubevirt import (
+    InventoryModule,
+)
+
+from ansible.template import Templar
+
+
+@pytest.fixture(scope="function")
+def inventory_composable_vars(mocker):
+    inventory = InventoryModule()
+
+    inventory.templar = Templar(loader=None)
+
+    inventory._options = {
+        "compose": {"block_migratable_vmis": "vmi_migration_method"},
+        "strict": True,
+        "groups": {"vmi_node_groups": "cluster_name"},
+        "keyed_groups": [{"prefix": "fedora", "key": "vmi_guest_os_info.version"}],
+    }
+
+    inventory.inventory = mocker.Mock()
+
+    return inventory
+
+
+@pytest.fixture(scope="function")
+def host_vars(monkeypatch, inventory_composable_vars):
+    host_vars = {}
+
+    def set_variable(host, key, value):
+        if host not in host_vars:
+            host_vars[host] = {}
+        host_vars[host][key] = value
+
+    monkeypatch.setattr(
+        inventory_composable_vars.inventory, "set_variable", set_variable
+    )
+    return host_vars
+
+
+@pytest.fixture(scope="function")
+def add_group(monkeypatch, inventory_composable_vars):
+    groups = []
+
+    def add_group(name):
+        if name not in groups:
+            groups.append(name)
+        return name
+
+    monkeypatch.setattr(inventory_composable_vars.inventory, "add_group", add_group)
+    return groups
+
+
+@pytest.fixture(scope="function")
+def add_host(monkeypatch, inventory_composable_vars):
+    hosts = []
+
+    def add_host(name, group=None):
+        if name not in hosts:
+            hosts.append(name)
+        if group is not None and group not in hosts:
+            hosts.append(group)
+
+    monkeypatch.setattr(inventory_composable_vars.inventory, "add_host", add_host)
+    return hosts
+
+
+@pytest.fixture(scope="function")
+def add_child(monkeypatch, inventory_composable_vars):
+    children = {}
+
+    def add_child(group, name):
+        if group not in children:
+            children[group] = []
+        if name not in children[group]:
+            children[group].append(name)
+
+    monkeypatch.setattr(inventory_composable_vars.inventory, "add_child", add_child)
+    return children
+
+
+@pytest.fixture(scope="module")
+def vmi_host_vars():
+    return {
+        "vmi_migration_method": "BlockMigration",
+        "vmi_guest_os_info": {"id": "fedora", "version": "39"},
+        "cluster_name": {"test-cluster"},
+    }
+
+
+@pytest.fixture(scope="module")
+def expected_host_vars():
+    return {"testvmi": {"block_migratable_vmis": "BlockMigration"}}
+
+
+@pytest.fixture(scope="module")
+def expected_groups():
+    return [
+        "vmi_node_groups",
+        "fedora_39",
+    ]
+
+
+@pytest.fixture(scope="module")
+def expected_children():
```

Comment:
I don't think we need separate fixtures here either. Just define/comparte them directly in the test.

Reply:
Done.